database-restore-postgres.sh

```
@@ -0,0 +1,289 @@
+}
+
+# Load environment variables from .backup file if present
+dir="$(dirname "$(realpath "$0")")"
+if [[ -f "$dir/.backup" ]]; then
+    export $(grep -v '^#' "$dir/.backup" | xargs)
+    log "Exported environment variables from .backup file"
+    
+    # Print exported variables for verification (masking password)
+    echo "DB_USERNAME=$DB_USERNAME"
+    echo "DB_PASSWORD=****"  # Mask password for security
+    echo "BACKUP_DIR=$BACKUP_DIR"
```

Shellcheck (0.10.0)

[info] Possible misspelling: BACKUP_DIR may not be assigned. Did you mean backup_dir?

(SC2153)